cypress/Dockerfile

```diff
@@ -1,64 +1,58 @@
-RUN yum -y install http://mirror.ghettoforge.org/distributions/gf/gf-release-latest.gf.el8.noarch.rpm
+# Update npm to latest version
+RUN npm install -g npm@latest
 
-# Install Cypress' dependencies
-COPY cypress/install-cypress-dependencies.sh /
-RUN /bin/sh install-cypress-dependencies.sh
+# Install necessary packages
+RUN apt-get update && apt-get install -y curl unzip && apt-get clean && rm -rf /var/lib/apt/lists/*
 
-# Cleanup vulnerabilities (as root user)
-RUN rm -rf /usr/local/share/.cache/yarn/v6/npm-public-encrypt-*/node_modules/public-encrypt/test/*
-RUN rm -rf /root/.cache/Cypress/**/Cypress/resources/app/node_modules/public-encrypt/test/*
-RUN rm -rf /root/.cache/Cypress/**/Cypress/resources/app/node_modules/http-proxy/test/*
-RUN rm -rf /root/.cache/Cypress/**/Cypress/resources/app/node_modules/lazystream/secret
+# Set working directory
+WORKDIR /hmda-frontend
 
 # Prepare working directory
-RUN mkdir -p hmda-frontend/cypress/downloads hmda-frontend/cypress/screenshots hmda-frontend/cypress/videos hmda-frontend/cypress/integration hmda-frontend/cypress/fixtures
-WORKDIR /hmda-frontend/
+RUN mkdir -p cypress/downloads cypress/screenshots cypress/videos cypress/integration cypress/fixtures
 
 # Copy Cypress support files
+COPY cypress/e2e cypress/e2e
 COPY cypress/support cypress/support
 COPY cypress/plugins cypress/plugins
 COPY cypress/docker-runner.sh cypress/
 COPY ./cypress.config.js ./
 
 # Copy HMDA helper functions
-RUN mkdir -p src/common
+RUN mkdir -p src/common src/filing/api src/filing/utils src/tools/larft/config
 COPY ../src/common/configUtils.js src/common/
 COPY ../src/deriveConfig.js src/
 COPY ../src/common/constants src/common/constants
-COPY ../src/common/api/ src/common/
-RUN mkdir -p src/filing/api/
+COPY ../src/common/api/ src/common/api/
 COPY ../src/filing/api src/filing/api
-RUN mkdir -p src/filing/utils/
 COPY ../src/filing/utils src/filing/utils
-RUN mkdir -p src/tools/larft/config/
 COPY ../src/tools/larft/config/ src/tools/larft/config/
 
+# Copy package.json
+COPY cypress/package.json ./
+
 # Enable creation of screenshots/videos
 RUN useradd -M -d /hmda-frontend -s /bin/bash hmda_cypress_user && \
   chown -R hmda_cypress_user:hmda_cypress_user /hmda-frontend/
 USER hmda_cypress_user
 
-# Install Cypress + supporting libraries
-COPY cypress/package.json ./
-RUN yarn install
+# Install dependencies and Cypress, then clean up vulnerable packages
+RUN npm install && \
+    npx cypress install --version ${CYPRESS_VERSION} && \
+    npx cypress verify
+
+USER root
+RUN find / -name "loader-utils" -type d -exec rm -rf {} + 2>/dev/null || true && \
+    find / -name "flat" -type d -exec rm -rf {} + 2>/dev/null || true && \
+    find / -name "underscore" -type d -exec rm -rf {} + 2>/dev/null || true && \
+    npm cache clean --force && \
+    chown -R hmda_cypress_user:hmda_cypress_user /hmda-frontend
```

Comment:
Removing critical CVE packages. Not all of them can be removed otherwise image will not run.